### PR TITLE
Add integration steps to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ericnordelo**
- Add the integration test step entry to CONTRIBUTING.md.
- Update PULL_REQUEST_TEMPLATE.md to suggest running Integration Tests and remove Changelog entry.

### CODE REVIEW

**Comment by ericnordelo** [comment]
This will close the Changelog entry issue in the PR Template by removing it.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -109,6 +109,12 @@ If you want to run the documentation UI locally:
     npm run docs:watch
     ```
 
+## Integration tests
+
+Currently, [starknet-devnet](https://github.com/Shard-Labs/starknet-devnet) is not enough as a source of truth to test smart contracts (behavior has important differences with testnet and mainnet). That's why we strongly suggest running some integration tests against the public testnet before submitting the PR, to make sure that everything works as expected in a real environment.
+
+We are working on defining a process for these integration tests, but for now, they must be run manually either by the contributor or/and the PR reviewer.
+
```
Comment (martriay) [inline CONTRIBUTING.md:117]:
```suggestion
## Integration tests

Currently, [starknet's test suite](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/testing/starknet.py) has important differences with public networks. Like [not checking signature hints toward the end of the tx flow](https://github.com/OpenZeppelin/cairo-contracts/issues/386).

That's why we strongly suggest testing new features against a testnet before submitting the PR, to make sure that everything works as expected in a real environment.

We are looking into defining a better process for these integration tests, but for now the PR author/contributor must suggest an approach to test the feature when applicable, which has to be agreed and reproduced by reviewer.
```
the devnet is just an http wrapper around the testing framework

**Review by martriay — COMMENTED** [review]
looking good!

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -15,6 +15,6 @@ Fixes #???? <!-- Fill in with issue number -->
 <!-- Feel free to submit a PR or Draft PR even if some items are pending. -->
 <!-- Some of the items may not apply. -->
 
-- [ ] Tests
+- [ ] Unit tests
+- [ ] Integration tests
```
[the remark this answers is not in the document]
Comment (ericnordelo) [reply PULL_REQUEST_TEMPLATE.md]: Agree. Updated.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -109,6 +109,14 @@ If you want to run the documentation UI locally:
     npm run docs:watch
     ```
 
+## Integration tests
+
+Currently, [starknet's test suite](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/testing/starknet.py) has important differences with public networks. Like [not checking signature hints toward the end of the tx flow](https://github.com/OpenZeppelin/cairo-contracts/issues/386).
+
+That's why we strongly suggest testing new features against a testnet before submitting the PR, to make sure that everything works as expected in a real environment.
+
+We are looking into defining a better process for these integration tests, but for now the PR author/contributor must suggest an approach to test the feature when applicable, which has to be agreed and reproduced by reviewer.
```
Comment (martriay) [inline CONTRIBUTING.md:118]:
```suggestion
We are looking into defining a better process for these integration tests, but for now the PR author/contributor must suggest an approach to test the feature when applicable, which has to be agreed and reproduced by the reviewer.
```

**Review by martriay — APPROVED** [review]

**Pull request merged**